protobuf-delta-lake-sink/src/main.rs

```diff
@@ -0,0 +1,156 @@
+    #[clap(long)]
+    pub source_region: String,
+
+    /// Should only be used for local testing
+    #[clap(long)]
+    pub source_access_key_id: Option<String>,
+    #[clap(long)]
+    pub source_secret_access_key: Option<String>,
+    #[clap(flatten)]
+    pub source_filter: FileFilter,
+
+    /// Url of the proto definition
+    #[clap(long)]
+    pub source_protos: Vec<String>,
+    /// Name of the proto message
+    #[clap(long)]
+    pub source_proto_name: String,
+
+    /// If provided, will use this column to partition by date
+    #[clap(long)]
+    pub partition_timestamp_column: Option<String>,
+
+    #[clap(long)]
+    pub target_bucket: String,
+    #[clap(long)]
+    pub target_table: String,
+    /// Optional api endpoint for the bucket. Default none
+    #[clap(long)]
+    pub target_endpoint: Option<String>,
+    /// Optional region for the endpoint. Default: us-west-2
+    #[clap(long)]
+    pub target_region: String,
+
+    /// Should only be used for local testing
+    #[clap(long)]
+    pub target_access_key_id: Option<String>,
+    #[clap(long)]
+    pub target_secret_access_key: Option<String>,
+
+    /// Maximum number of records to write at a time. Controls the size of outputted parquet files, and memory usage.
+    #[clap(long, default_value = "1000000")]
+    pub max_records: usize,
+}
+
+#[tokio::main]
+async fn main() {
+    use clap::Parser;
+    let args = Args::parse();
+
+    let descriptor = &get_descriptor(args.source_protos, args.source_proto_name).await;
+    let delta_schema = get_delta_schema(&descriptor, args.partition_timestamp_column.is_some());
+
+    let mut s3_config =
+        HashMap::from([("aws_default_region".to_string(), args.target_region.clone())]);
+    if let Some(aws_secret_key_id) = args.target_secret_access_key {
+        s3_config.insert("aws_secret_access_key".to_string(), aws_secret_key_id);
+        s3_config.insert("allow_http".to_string(), "true".to_string());
+    }
+    if let Some(access_key_id) = args.target_access_key_id {
+        s3_config.insert("aws_access_key_id".to_string(), access_key_id);
+    }
+    if let Some(endpoint) = args.target_endpoint {
+        s3_config.insert("aws_endpoint".to_string(), endpoint);
+    }
+
+    let uri = format!("s3://{}/{}", args.target_bucket, args.target_table);
+    let mut table_raw = DeltaTableBuilder::from_uri(uri)
+        .with_storage_options(s3_config)
+        .build()
+        .unwrap();
+
+    let maybe_table = table_raw.load().await;
+
+    let mut table = match maybe_table {
+        Ok(_) => table_raw,
+        Err(DeltaTableError::NotATable(_)) => {
+            println!("It doesn't look like our delta table has been created");
+            DeltaOps(table_raw.into())
+                .create()
+                .with_columns(delta_schema)
+                .with_partition_columns(vec!["date"])
+                .await
+                .unwrap()
+        }
+        Err(err) => Err(err).unwrap(),
```

Comment:
Is this to panic on the error? tokio main functions are allowed to return `Result<()>`

Reply:
Just had it return result and used anyhow to coerce everything. Anyhow is super nice